hack/patches/k8s---issue128.patch

```diff
@@ -0,0 +1,141 @@
++	kl.SetCadvisorClient(cadvisorClient)
+ }
+ 
+-// TODO: move this into the kubelet itself
+-func InitHealthChecking(k *Kubelet) {
++func (kl *Kubelet) InitHealthChecking() {
+ 	// TODO: These should probably become more plugin-ish: register a factory func
+ 	// in each checker's init(), iterate those here.
+-	health.AddHealthChecker(health.NewExecHealthChecker(k))
++	health.AddHealthChecker(health.NewExecHealthChecker(kl))
+ 	health.AddHealthChecker(health.NewHTTPHealthChecker(&http.Client{}))
+ 	health.AddHealthChecker(&health.TCPHealthChecker{})
+ }
+diff --git a/pkg/standalone/standalone.go b/pkg/standalone/standalone.go
+index b9af29a..ff5bc4b 100644
+--- a/pkg/standalone/standalone.go
++++ b/pkg/standalone/standalone.go
+@@ -143,15 +143,29 @@ func SimpleRunKubelet(etcdClient tools.EtcdClient, dockerClient dockertools.Dock
+ 		EnableDebuggingHandlers: true,
+ 		SyncFrequency:           3 * time.Second,
+ 	}
+-	RunKubelet(&kcfg)
++	RunKubelet(&kcfg, nil)
+ }
+ 
++// bootstrapping interface for kubelet, targets the initialization protocol
++type KubeletBootstrap interface {
++	BirthCry()
++	GarbageCollectLoop()
++	InitHealthChecking()
++	ListenAndServe(net.IP, uint, bool)
++	MonitorCAdvisor(uint)
++	Run(<-chan kubelet.PodUpdate)
++	RunOnce(<-chan kubelet.PodUpdate) ([]kubelet.RunPodResult, error)
++}
++
++// create and initialize a Kubelet instance
++type KubeletBuilder func(kc *KubeletConfig) (KubeletBootstrap, *config.PodConfig)
++
+ // RunKubelet is responsible for setting up and running a kubelet.  It is used in three different applications:
+ //   1 Integration tests
+ //   2 Kubelet binary
+ //   3 Standalone 'kubernetes' binary
+ // Eventually, #2 will be replaced with instances of #3
+-func RunKubelet(kcfg *KubeletConfig) {
++func RunKubelet(kcfg *KubeletConfig, builder KubeletBuilder) {
+ 	kubelet.SetupEventSending(kcfg.AuthPath, kcfg.ApiServerList)
+ 	kubelet.SetupLogging()
+ 	kubelet.SetupCapabilities(kcfg.AllowPrivileged)
+@@ -161,8 +175,10 @@ func RunKubelet(kcfg *KubeletConfig) {
+ 		kubelet.SetupRootDirectoryOrDie(kcfg.RootDirectory)
+ 	}
+ 
+-	cfg := makePodSourceConfig(kcfg)
+-	k := createAndInitKubelet(kcfg, cfg)
++	if builder == nil {
++		builder = createAndInitKubelet
++	}
++	k, cfg := builder(kcfg)
+ 	// process pods and exit.
+ 	if kcfg.Runonce {
+ 		if _, err := k.RunOnce(cfg.Updates()); err != nil {
+@@ -173,15 +189,15 @@ func RunKubelet(kcfg *KubeletConfig) {
+ 	}
+ }
+ 
+-func startKubelet(k *kubelet.Kubelet, cfg *config.PodConfig, kc *KubeletConfig) {
++func startKubelet(k KubeletBootstrap, cfg *config.PodConfig, kc *KubeletConfig) {
+ 	// start the kubelet
+ 	go util.Forever(func() { k.Run(cfg.Updates()) }, 0)
+ 
+ 	// start the kubelet server
+ 	if kc.EnableServer {
+ 		go util.Forever(func() {
+-			kubelet.ListenAndServeKubeletServer(k, net.IP(kc.Address), kc.Port, kc.EnableDebuggingHandlers)
+-		}, 0)
++			k.ListenAndServe(net.IP(kc.Address), kc.Port, kc.EnableDebuggingHandlers)
++		}, 10*time.Second)
```

Comment:
Maybe add the "We don't want 0s because if we can't get the port right away then we generate **tons** of logging and fill up the disk." comment here?